R/TealAppDriver.R

```diff
@@ -311,39 +311,70 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
       active_filters[[dataset_name]]
     },
     #' @description
-    #' Add a new variable from the dataset to be filtered.
+    #' Adds a new filter variable to the dataset. Supports both standard datasets and MAE datasets.
     #'
     #' @param dataset_name (character) The name of the dataset to add the filter variable to.
+    #' @param sample_name (character) The name of the sample/experiment.
     #' @param var_name (character) The name of the variable to add to the filter panel.
+    #' @param filter_type (character) Specifies the type of filter. Default is "data.frame".
     #' @param ... Additional arguments to be passed to `shinytest2::AppDriver$set_inputs`
     #'
     #' @return The `TealAppDriver` object invisibly.
-    add_filter_var = function(dataset_name, var_name, ...) {
+    add_filter_var = function(dataset_name, sample_name, var_name, filter_type = "data.frame", ...) {
```

Comment:
I think  setting  specific default value (e.g., "data.frame" for filter_type) rather than NULL makes the function's intended use and behaviour more clearer to the user.  we internally anyways going to change it 'data.frame`. It communicates what the expected default behavior in definition  without requiring the user to dive into the function's logic.